### ISSUE TITLE
Writing other than single bits or single bytes is buggy..

### ISSUE THREAD

**Issue by Tobias004** [ShadowDancer/BitStreams#2]
or do I miss something? Since the class ``BitStream`` inherits from ``Stream`` all methods should be usable.

```csharp
            using (FileStream f = new FileStream("Test.dat", FileMode.OpenOrCreate, FileAccess.ReadWrite)) {
                BitStream stream = new BitStream(BitDirection.LsbFirst,f);
                stream.WriteBit(true);
                stream.Write(BitConverter.GetBytes((ushort)0xAAAA));
                stream.Flush();
             }
            using (FileStream f = new FileStream("Test.dat", FileMode.OpenOrCreate, FileAccess.ReadWrite)) {
                BitStream stream = new BitStream(BitDirection.LsbFirst,f);
                bool bit = stream.ReadBitAsBool();
                Span<byte> bytes = stackalloc byte[2];
                stream.Read(bytes);
                ulong us = BitConverter.ToUInt16(bytes);
                //us == 0x00AA  but should be 0xAAAA
                //Hexdump of Test.dat: 55 55 01
            }
```

The terms "MSB" and  "LSB" are initially somewhat misleading for me. They should represent the byte order of words. For bits high-order bit or left-most bit is more clear for me.
I think finally it's better to overwrite ``BinaryReader`` and ``BinaryWriter`` .

**Comment by ShadowDancer** [comment]
Hi,
I think your sample should work, apparently there is bug in the library. I'll take a look later this weekend.


```The terms "MSB" and "LSB" are initially somewhat misleading for me. They should represent the byte order of words. For bits high-order bit or left-most bit is more clear for me.```
> I think it depends on your background. I've added these aliases to docs, so it's not necessary to google it up. These stand for "Most significant bit" and "Least significant bit" and I think these are fairly common in literature (see Bit numbering on wikipedia). 

```I think finally it's better to overwrite BinaryReader and BinaryWriter .```
I dislike this idea because BinaryReader/Writer do not have concept of endianness. One possibility would be to create class like BinaryReader, which would have only useful methods.
Overriding stream has problem of multiple flushes, i.e.:
```
                stream.WriteBit(true);
                stream.Flush();
                stream.WriteBit(true);
                stream.Flush();
```
But when you think about it `BitWriter` implementation would also have this problem, because program have to do something with "dangling bit".

**Comment by Tobias004** [comment]
It's great that you respond so fast, many thanks!
Update:
I tested it again with a .NET 5.0 console project (your lib via nuget),
and got an IndexOutOfRangeException when trying to read the two bytes
( 
btw. stream.ReadBitAsBool() is my simple extension:
```csharp
            int bit = ReadBit();
            if (bit == -1)
                throw new EndOfStreamException();
            return bit > 0;
```

**Comment by ShadowDancer** [comment]
There was bug related to reading multiple bytes when bit buffer was not empty.
I think these bugs happened because this is not regular use case (at least for me :)).

It should be fixed in version 0.0.5.
